subiquitycore/ui/views/hostname.py

```diff
@@ -0,0 +1,122 @@
+from subiquitycore.view import BaseView
+
+log = logging.getLogger("subiquitycore.views.hostname")
+
+
+'''
++----------------------------------------------+
+|                                              |
+| Enter the hostname to use for the device     |
+|                                              |
+|              +-------------------------+     |
+|    Hostname: | localhost               |     |
+|              +-------------------------+     |
+|                                              |
+|                                              |
+|                         +--------+           |
+|                         | Done   |           |
+|                         +--------+           |
+|                         | Cancel |           |
+|                         +--------+           |
+|                                              |
++----------------------------------------------+
+'''
+
+class SubmittingHostnameEditor(StringEditor):
+
+    def __init__(self, mainview):
+        self.mainview = mainview
+        super().__init__(caption="")
+
+    def keypress(self, size, key):
+        if key == 'enter':
+            self.mainview.done(None)
+            return None
+        else:
+            return super().keypress(size, key)
+
+
+class HostnameView(BaseView):
+
+    def __init__(self, model, signal, opts):
+        self.model = model
+        self.signal = signal
+        self.opts = opts
+        self.items = []
+        self.hostname = SubmittingHostnameEditor(self)
+        self.hostname.value = socket.gethostname()
+        self.error = Text("", align="center")
+
+        body = [
+            Padding.center_90(self._build_model_inputs()),
+            Padding.line_break(""),
+            Padding.center_90(Color.info_error(self.error)),
+            Padding.line_break(""),
+            Padding.fixed_10(self._build_buttons()),
+        ]
+        super().__init__(ListBox(body))
+
+    def _build_model_inputs(self):
+        sl = [
+            Columns(
+                [
+                    ("weight", 0.2, Text("Hostname:", align="right")),
+                    ("weight", 0.3,
+                     Color.string_input(self.hostname,
+                                        focus_map="string_input focus"))
+                ],
+                dividechars=4
+            ),
+        ]
+        return Pile(sl)
+
+    def _build_buttons(self):
+        cancel = cancel_btn(on_press=self.cancel)
+        done = done_btn(on_press=self.done)
+
+        buttons = [
+            Color.button(done, focus_map='button focus'),
+            Color.button(cancel, focus_map='button focus')
+        ]
+        return Pile(buttons)
+
+    def cancel(self, button):
+        self.signal.prev_signal()
+
+    def done(self, button):
+        if len(self.hostname.value) < 1:
+            self.error.set_text("Please enter a non-empty name.")
+            return
+        if not self.opts.dry_run:
+            cmd = ["hostname", self.hostname.value]
```

Comment:
Ah yeah, you're right. Is editing /etc/hosts really necessary though? After running hostname foo.local on in my KVM test system I can resolve foo.local. (Maybe this is some resolved magic?)